### PR TITLE
ff: core algorithms for BitSplit

### PR DESCRIPTION
**Pull request by alex-ozdemir**

### CODE REVIEW

File: test/unit/theory/theory_ff_split_gb_black.cpp [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+  void SetUp() override
+  {
+    TestSmt::SetUp();
+    initCocoaGlobalManager();
+  }
+};
+
+CoCoA::RingElem randCoeff(const CoCoA::ring& polyRing)
+{
+  return CoCoA::zero(CoCoA::CoeffRing(polyRing)) + rand();
```
Comment (HanielB) [inline test/unit/theory/theory_ff_split_gb_black.cpp:52]: for reproducibility, don't we want to use here `Random` from `util/random.h`?

File: test/unit/theory/theory_ff_split_gb_black.cpp [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+}
+
+CoCoA::RingElem randPoly(const CoCoA::ring& polyRing,
+                         size_t degree,
+                         size_t terms)
+{
+  CoCoA::RingElem out = CoCoA::zero(polyRing);
+  // CoCoA::RingElem out = CoCoA::zero(polyRing) + randCoeff(polyRing);
```
Comment (HanielB) [inline test/unit/theory/theory_ff_split_gb_black.cpp:60]: clean

File: test/unit/theory/theory_ff_split_gb_black.cpp [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+                                 std::vector<CoCoA::RingElem> root)
+{
+  CoCoA::RingElem p = randPoly(polyRing, degree, terms);
+  CoCoA::RingElem val = ff::cocoaEval(p, root);
+  return p - val;
+}
+
+TEST_F(TestTheoryFfSplitGb, RandSat)
+{
+  // TraceChannel.on("ff::split::mc");
```
Comment (HanielB) [inline test/unit/theory/theory_ff_split_gb_black.cpp:87]: ditto. Several other cases also.

**Comment by alex-ozdemir** [comment]
Thanks for the feedback! Let me know if you have more.

**Review by HanielB — APPROVED** [review]
Looks good to me now, thanks!

**Pull request merged**